### ISSUE TITLE
is there a way I can use this plugin with mono

### ISSUE THREAD

**Issue by Vad1mo** [Ullink/gradle-ikvm-plugin#7]
Hello I am using mono on Mac/Linux CI and would like to use this plugin and replace my script.

In Mono we use ikvm like this 

```mono ikvm-8.0.5449.1/bin/ikvmc.exe ..... ````
so what we need in this plugin is an option so say I am using mono an prepend the mono command before execution. 

is this supported or would that be possible?

**Comment by gluck** [comment]
Mono support isn't yet in the plugin, but it'd be fairly easy to add:
https://github.com/Ullink/gradle-ikvm-plugin/blob/master/src/main/groovy/com/ullink/Ikvm.groovy#L61

Would you feel ready to contribute it ?

**Comment by Vad1mo** [comment]
will this do the trick?

```
    @InputFile
    def getIkvmc(){
        assert getIkvmHome(), "You must install Ikvm and set ikvm.home property or IKVM_HOME env variable"
        File ikvmExec = new File(project.file(getIkvmHome()), 'bin/ikvmc.exe')
        assert ikvmExec.exists(), "You must install Ikvm and set ikvm.home property or IKVM_HOME env variable"
        if (System.getProperty("os.name") != "win"){
            return "mono " +  ikvmExec
        }
        return ikvmExec
    } 
```

**Comment by gluck** [comment]
Almost !
- use `OperatingSystem.current().windows` instead of the property
- return an array ["mono", ikvmExec], as I don't think the string with space will work
- the method annotation @InputFile will probably not work on 'mono' file, so you should add a new Method (w/o the annotation) that'll return either ikvmc or mono+ikvmc

**Comment by Vad1mo** [comment]
thank you.

I have some difficulties testing the plugin in my test project.

I published the plugin to my local maven repo with `gradle publishPluginPublicationToMavenLocal` and it is located in `/Users/v/.m2/repository/com/ullink/gradle/gradle-ikvm-plugin/2.3/gradle-ikvm-plugin-2.3.jar`
as expected.

```
buildscript {
    repositories {
        mavenLocal()
        mavenCentral()
    }
    dependencies {
        classpath "com.ullink.gradle:gradle-ikvm-plugin:2.3" //this works
    }
}

plugins {
   id 'java'
   id 'eclipse'
   id 'com.github.johnrengelman.shadow' version '1.2.1'
   //id "com.ullink.ikvm" version "2.3"
}
apply plugin:'ikvm' // here is the error Failed to apply plugin [id 'ikvm']  Could not create task of type 'Ikvm'.
```

**Comment by gluck** [comment]
Running with --info and/or --stacktrace will probably give you a better error message/cause.

**Comment by Vad1mo** [comment]
no

```
> Failed to apply plugin [id 'ikvm']
   > Could not create task of type 'Ikvm'.

* Try:
Run with --info or --debug option to get more log output.

* Exception is:
org.gradle.api.GradleScriptException: A problem occurred evaluating root project 'ZUGFeRD.NET'.
	at org.gradle.groovy.scripts.internal.DefaultScriptRunnerFactory$ScriptRunnerImpl.run(DefaultScriptRunnerFactory.java:54)
	at org.gradle.configuration.DefaultScriptPluginFactory$ScriptPluginImpl.apply(DefaultScriptPluginFactory.java:154)
	at org.gradle.configuration.project.BuildScriptProcessor.execute(BuildScriptProcessor.java:39)
	at org.gradle.configuration.project.BuildScriptProcessor.execute(BuildScriptProcessor.java:26)
	at org.gradle.configuration.project.ConfigureActionsProjectEvaluator.evaluate(ConfigureActionsProjectEvaluator.java:34)
	at org.gradle.configuration.project.LifecycleProjectEvaluator.evaluate(LifecycleProjectEvaluator.java:59)
	at org.gradle.api.internal.project.AbstractProject.evaluate(AbstractProject.java:492)
	at org.gradle.api.internal.project.AbstractProject.evaluate(AbstractProject.java:86)
	at org.gradle.execution.TaskPathProjectEvaluator.configureHierarchy(TaskPathProjectEvaluator.java:42)
	at org.gradle.configuration.DefaultBuildConfigurer.configure(DefaultBuildConfigurer.java:35)
	at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:129)
	at org.gradle.initialization.DefaultGradleLauncher.doBuild(DefaultGradleLauncher.java:106)
	at org.gradle.initialization.DefaultGradleLauncher.run(DefaultGradleLauncher.java:86)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter$DefaultBuildController.run(InProcessBuildActionExecuter.java:80)
	at org.gradle.launcher.cli.ExecuteBuildAction.run(ExecuteBuildAction.java:33)
	at org.gradle.launcher.cli.ExecuteBuildAction.run(ExecuteBuildAction.java:24)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:36)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:26)
	at org.gradle.launcher.daemon.server.exec.ExecuteBuild.doBuild(ExecuteBuild.java:48)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.WatchForDisconnection.execute(WatchForDisconnection.java:37)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.ResetDeprecationLogger.execute(ResetDeprecationLogger.java:26)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.RequestStopIfSingleUsedDaemon.execute(RequestStopIfSingleUsedDaemon.java:34)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:74)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:72)
	at org.gradle.util.Swapper.swap(Swapper.java:38)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput.execute(ForwardClientInput.java:72)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.health.DaemonHealthTracker.execute(DaemonHealthTracker.java:47)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.LogToClient.doBuild(LogToClient.java:60)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment.doBuild(EstablishBuildEnvironment.java:71)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.health.HintGCAfterBuild.execute(HintGCAfterBuild.java:41)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy$1.run(StartBuildOrRespondWithBusy.java:50)
	at org.gradle.launcher.daemon.server.DaemonStateCoordinator$1.run(DaemonStateCoordinator.java:246)
	at org.gradle.internal.concurrent.DefaultExecutorFactory$StoppableExecutorImpl$1.run(DefaultExecutorFactory.java:64)
Caused by: org.gradle.api.internal.plugins.PluginApplicationException: Failed to apply plugin [id 'ikvm']
	at org.gradle.api.internal.plugins.DefaultPluginManager.doApply(DefaultPluginManager.java:147)
	at org.gradle.api.internal.plugins.DefaultPluginManager.apply(DefaultPluginManager.java:102)
	at org.gradle.api.internal.plugins.DefaultObjectConfigurationAction.applyType(DefaultObjectConfigurationAction.java:113)
	at org.gradle.api.internal.plugins.DefaultObjectConfigurationAction.access$200(DefaultObjectConfigurationAction.java:36)
	at org.gradle.api.internal.plugins.DefaultObjectConfigurationAction$3.run(DefaultObjectConfigurationAction.java:80)
	at org.gradle.api.internal.plugins.DefaultObjectConfigurationAction.execute(DefaultObjectConfigurationAction.java:136)
	at org.gradle.api.internal.project.AbstractPluginAware.apply(AbstractPluginAware.java:46)
	at org.gradle.api.plugins.PluginAware$apply.call(Unknown Source)
	at org.gradle.api.internal.project.ProjectScript.apply(ProjectScript.groovy:34)
	at org.gradle.api.Script$apply.callCurrent(Unknown Source)
	at build_16s0iada1nmeyzh89jc0y7l79.run(/Users/vadim/Development/konikproject/konik.net/ZUGFeRD.NET/build.gradle:17)
	at org.gradle.groovy.scripts.internal.DefaultScriptRunnerFactory$ScriptRunnerImpl.run(DefaultScriptRunnerFactory.java:52)
	... 44 more
Caused by: org.gradle.api.tasks.TaskInstantiationException: Could not create task of type 'Ikvm'.
	at org.gradle.api.internal.project.taskfactory.TaskFactory$1.call(TaskFactory.java:126)
	at org.gradle.api.internal.project.taskfactory.TaskFactory$1.call(TaskFactory.java:121)
	at org.gradle.api.internal.AbstractTask.injectIntoNewInstance(AbstractTask.java:139)
	at org.gradle.api.internal.project.taskfactory.TaskFactory.createTaskObject(TaskFactory.java:121)
	at org.gradle.api.internal.project.taskfactory.TaskFactory.createTask(TaskFactory.java:81)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory.createTask(AnnotationProcessingTaskFactory.java:99)
	at org.gradle.api.internal.project.taskfactory.DependencyAutoWireTaskFactory.createTask(DependencyAutoWireTaskFactory.java:39)
	at org.gradle.api.internal.tasks.DefaultTaskContainer.create(DefaultTaskContainer.java:52)
	at org.gradle.api.internal.project.AbstractProject.task(AbstractProject.java:859)
	at org.gradle.api.Project$task.call(Unknown Source)
	at com.ullink.IkvmPlugin.apply(IkvmPlugin.groovy:27)
	at com.ullink.IkvmPlugin.apply(IkvmPlugin.groovy)
	at org.gradle.api.internal.plugins.ImperativeOnlyPluginApplicator.applyImperative(ImperativeOnlyPluginApplicator.java:35)
	at org.gradle.api.internal.plugins.RulesCapablePluginApplicator.applyImperative(RulesCapablePluginApplicator.java:42)
	at org.gradle.api.internal.plugins.DefaultPluginManager.doApply(DefaultPluginManager.java:133)
	... 55 more
Caused by: java.lang.NoClassDefFoundError: org/apache/commons/lang/StringUtils
	at com.ullink.Ikvm.getCompileConfigurationName(Ikvm.groovy:58)
	at com.ullink.Ikvm_Decorated.getCompileConfigurationName(Unknown Source)
	at com.ullink.Ikvm.<init>(Ikvm.groovy:49)
	at com.ullink.Ikvm_Decorated.<init>(Unknown Source)
	at org.gradle.api.internal.DependencyInjectingInstantiator.newInstance(DependencyInjectingInstantiator.java:48)
	at org.gradle.api.internal.ClassGeneratorBackedInstantiator.newInstance(ClassGeneratorBackedInstantiator.java:36)
	at org.gradle.api.internal.project.taskfactory.TaskFactory$1.call(TaskFactory.java:124)
	... 69 more
Caused by: java.lang.ClassNotFoundException: org.apache.commons.lang.StringUtils
	... 76 more
```

**Comment by Vad1mo** [comment]
the plugin is on the class path because if I change
`classpath "com.ullink.gradle:gradle-ikvm-plugin:2.3"` to something different lets say 2.4 gradle is complaining that the jar/pom was not found.

**Comment by Vad1mo** [comment]
sorry i got that now, the plugin hat issues during loading. But now I have a question  at com.ullink.Ikvm.getCompileConfigurationName(Ikvm.groovy:58) StringUtils is used 
but should I import this dependency in my test project? This would be strange if I had to redeclare all the dependencies.

**Comment by Vad1mo** [comment]
this will do it there was to much on the classpath that wasn't included. I wonder how it did work in the past.

```    String getCompileConfigurationName() {
        return String.format("%sCompile", this.name.toLowerCase());
    }
```

**Comment by gluck** [comment]
Released 2.3 version, thx ! :shipit:
